# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: 3cfa16c9f9a9
# Full Hash: 3cfa16c9f9a97c5963b31a9ca080df29e772836c
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
# ==============================================================================

diff -r 50772b4ff6aa -r 3cfa16c9f9a9 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Fri Oct 23 18:17:37 2020 +0000
@@ -4411,6 +4411,9 @@
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
 
+wasm::TlsData* ExtractCalleeTlsFromFrameWithTls(wasm::Frame* fp);
+wasm::TlsData* ExtractCallerTlsFromFrameWithTls(wasm::Frame* fp);
+
 }  // namespace jit
 }  // namespace js
 